Makefile

```diff
@@ -41,45 +41,41 @@ YELLOW := $(shell tput -Txterm setaf 3)
 WHITE  := $(shell tput -Txterm setaf 7)
 RESET  := $(shell tput -Txterm sgr0)
 
-## Display help for all targets
-help:
-	@awk '/^[a-zA-Z_0-9%:\\\/-]+:/ { \
-		msg = match(lastLine, /^## (.*)/); \
-			if (msg) { \
-				cmd = $$1; \
-				msg = substr(lastLine, RSTART + 3, RLENGTH); \
-				gsub("\\\\", "", cmd); \
-				gsub(":+$$", "", cmd); \
-				printf "  \x1b[32;01m%-35s\x1b[0m %s\n", cmd, msg; \
-			} \
-	} \
-	{ lastLine = $$0 }' $(MAKEFILE_LIST) | sort -u
+.PHONY: default
+default: help
 
+.PHONY: build
 ## Build the docker image
-docker/build:
+build:
 	@docker build -t ${DOCKER_HUB_REPO}:latest -t ${DOCKER_HUB_REPO}:${DOCKER_IMAGE_TAG} .
 
+.PHONY: tag
 ## Create a new tag ( expects the "DOCKER_IMAGE_ADDITIONAL_TAG" environment variable to be set )
-docker/tag:
+tag:
 	@docker tag ${DOCKER_HUB_REPO}:${DOCKER_IMAGE_TAG} ${DOCKER_HUB_REPO}:${DOCKER_IMAGE_ADDITIONAL_TAG}
 
+.PHONY: save
 ## Save the docker image to disk
-docker/save:
+save:
 	@docker save ${DOCKER_HUB_REPO}:${DOCKER_IMAGE_TAG} > "${DOCKER_IMAGE_TAG}.tar"
 
+.PHONY: load
 ## Load saved image
-docker/load:
+load:
 	@docker load < "${DOCKER_IMAGE_TAG}.tar"
 
+.PHONY: login
```

Comment:
Hmm, not a big fan of this either. I think that bundling this and saving some more lines of code is the way to go or at least my personal preference. Your IDE should be able to tell you when a target is marked as a phony target and when it isn't. This is at least how I configured it in my environment. As said, this is my personal preference and I leave the decision to you :)

Reply:
not using a crazy IDE...and we already faced the issue of renaming run-tests to unit-tests without changing the .PHONY target.. ;) also this way it will create less potential merge conflicts... as changes are in different areas of the file